Plugin breaks non-JS/TS code

I'd love to use this to bundle resources that may include Deno code *and* non-JS code (like CSS/SCSS files). 

But adding the plugin to my ESBuild configuration breaks if there are files that aren't TypeScript/JavaScript.

IMO, if this plugin doesn't know how to handle these files, it should just skip them and allow other plugins (or esbuild itself) to handle them.

Reproduction 
--------------

I've got a minimal reproduction here:

https://github.com/NfNitLoop/deno-loader-bug

With a config like this:

```typescript
import * as esbuild from "npm:esbuild@0.20.2";
import { denoPlugins } from "jsr:@luca/esbuild-deno-loader@^0.10.3";

const result = await esbuild.build({
  plugins: [
    // ...denoPlugins()
  ],
  entryPoints: [
    "src/mod.ts",
    "src/style.css",
  ],
  outdir: "dist",
  bundle: true,
  format: "esm",
});
esbuild.stop();
```

This works, but if you uncomment the `denoPlugins()` line you get errors:

> ✘ [ERROR] Expected a JavaScript or TypeScript module, but identified a Unknown module. Importing these types of modules is currently not supported.
>  Specifier: file:///Users/codyc/code/bugs/deno-loader-bug/src/style.css [plugin deno-loader]

--- a/mod_test.ts
+++ b/mod_test.ts
@@ -884,6 +884,42 @@ Deno.test("custom plugin for scheme with import map", async (t) => {
   });
 });
 
+const TXT_PLUGIN: esbuildNative.Plugin = {
+  name: "computed",
+  setup(build) {
+    build.onLoad({ filter: /.*\.txt$/, namespace: "file" }, async (args) => {
+      const url = esbuildResolutionToURL(args);
+      const file = await Deno.readTextFile(new URL(url));
+      return {
+        contents: `export default ${JSON.stringify(file)};`,
+        loader: "js",
+      };
+    });
+  },
+};
+
+Deno.test("txt plugin", async (t) => {
+  +await testLoader(t, LOADERS, async (esbuild, loader) => {
+    const res = await esbuild.build({
+      ...DEFAULT_OPTS,
+      plugins: [
+        denoResolverPlugin(),
+        TXT_PLUGIN,
+        denoLoaderPlugin({ loader }),
+      ],
+      entryPoints: ["./testdata/hello.txt"],
+    });
+    assertEquals(res.warnings, []);
+    assertEquals(res.errors, []);
+    assertEquals(res.outputFiles.length, 1);
+    const output = res.outputFiles[0];
+    assertEquals(output.path, "<stdout>");
+    const dataURL = `data:application/javascript;base64,${btoa(output.text)}`;
+    const { default: hello } = await import(dataURL);
+    assertEquals(hello, "Hello World!");
+  });
+});
+
 Deno.test("uncached data url", async (t) => {
   await testLoader(t, LOADERS, async (esbuild, loader) => {
     const configPath = join(Deno.cwd(), "testdata", "config_ref.json");

--- a/src/esbuild_types.ts
+++ b/src/esbuild_types.ts
@@ -62,7 +62,7 @@ export interface PluginBuild {
   /** Documentation: https://esbuild.github.io/plugins/#on-load */
   onLoad(
     options: OnLoadOptions,
-    callback: (args: OnLoadArgs) => Promise<OnLoadResult | null> | undefined,
+    callback: (args: OnLoadArgs) => Promise<OnLoadResult | null | undefined> | undefined,
   ): void;
 }
 

--- a/src/esbuild_types.ts
+++ b/src/esbuild_types.ts
@@ -62,7 +62,9 @@ export interface PluginBuild {
   /** Documentation: https://esbuild.github.io/plugins/#on-load */
   onLoad(
     options: OnLoadOptions,
-    callback: (args: OnLoadArgs) => Promise<OnLoadResult | null | undefined> | undefined,
+    callback: (
+      args: OnLoadArgs,
+    ) => Promise<OnLoadResult | null | undefined> | undefined,
   ): void;
 }
 

--- a/src/loader_native.ts
+++ b/src/loader_native.ts
@@ -8,6 +8,7 @@ import {
   type Loader,
   type LoaderResolution,
   mapContentType,
+  mediaTypeFromSpecifier,
   mediaTypeToLoader,
   parseNpmSpecifier,
 } from "./shared.ts";
@@ -45,7 +46,15 @@ export class NativeLoader implements Loader {
     }
 
     const entry = await this.#infoCache.get(specifier.href);
-    if ("error" in entry) throw new Error(entry.error);
+    if ("error" in entry) {
+      if (
+        specifier.protocol === "file:" &&
+        mediaTypeFromSpecifier(specifier) === "Unknown"
+      ) {
+        return { kind: "esm", specifier: new URL(entry.specifier) };
+      }
+      throw new Error(entry.error);
+    }
 
     if (entry.kind === "npm") {
       // TODO(lucacasonato): remove parsing once https://github.com/denoland/deno/issues/18043 is resolved
@@ -66,23 +75,28 @@ export class NativeLoader implements Loader {
     return { kind: "esm", specifier: new URL(entry.specifier) };
   }
 
-  async loadEsm(specifier: URL): Promise<esbuild.OnLoadResult> {
+  async loadEsm(specifier: URL): Promise<esbuild.OnLoadResult | undefined> {
     if (specifier.protocol === "data:") {
       const resp = await fetch(specifier);
       const contents = new Uint8Array(await resp.arrayBuffer());
       const contentType = resp.headers.get("content-type");
       const mediaType = mapContentType(specifier, contentType);
       const loader = mediaTypeToLoader(mediaType);
+      if (loader === null) return undefined;
       return { contents, loader };
     }
     const entry = await this.#infoCache.get(specifier.href);
-    if ("error" in entry) throw new Error(entry.error);
+    if (
+      "error" in entry && specifier.protocol !== "file:" &&
+      mediaTypeFromSpecifier(specifier) !== "Unknown"
+    ) throw new Error(entry.error);
 
     if (!("local" in entry)) {
       throw new Error("[unreachable] Not an ESM module.");
     }
     if (!entry.local) throw new Error("Module not downloaded yet.");
     const loader = mediaTypeToLoader(entry.mediaType);
+    if (loader === null) return undefined;
 
     let contents = await Deno.readFile(entry.local);
     const denoCacheMetadata = lastIndexOfNeedle(contents, DENO_CACHE_METADATA);

--- a/src/loader_portable.ts
+++ b/src/loader_portable.ts
@@ -144,7 +144,7 @@ export class PortableLoader implements Loader, Disposable {
     );
   }
 
-  async loadEsm(url: URL): Promise<esbuild.OnLoadResult> {
+  async loadEsm(url: URL): Promise<esbuild.OnLoadResult | undefined> {
     let module: Module;
     switch (url.protocol) {
       case "file:": {
@@ -162,6 +162,7 @@ export class PortableLoader implements Loader, Disposable {
     }
 
     const loader = mediaTypeToLoader(module.mediaType);
+    if (loader === null) return undefined;
 
     const res: esbuild.OnLoadResult = { contents: module.data, loader };
     if (url.protocol === "file:") {

--- a/src/plugin_deno_loader.ts
+++ b/src/plugin_deno_loader.ts
@@ -375,7 +375,7 @@ export function denoLoaderPlugin(
 
       function onLoad(
         args: esbuild.OnLoadArgs,
-      ): Promise<esbuild.OnLoadResult | null> | undefined {
+      ): Promise<esbuild.OnLoadResult | null | undefined> | undefined {
         if (args.namespace === "file" && isInNodeModules(args.path)) {
           // inside node_modules, just let esbuild do it's thing
           return undefined;

--- a/src/shared.ts
+++ b/src/shared.ts
@@ -6,7 +6,7 @@ import type * as esbuild from "./esbuild_types.ts";
 
 export interface Loader {
   resolve(specifier: URL): Promise<LoaderResolution>;
-  loadEsm(specifier: URL): Promise<esbuild.OnLoadResult>;
+  loadEsm(specifier: URL): Promise<esbuild.OnLoadResult | undefined>;
 
   packageIdFromNameInPackage?(
     name: string,
@@ -39,7 +39,7 @@ export interface LoaderResolutionNode {
   path: string;
 }
 
-export function mediaTypeToLoader(mediaType: MediaType): esbuild.Loader {
+export function mediaTypeToLoader(mediaType: MediaType): esbuild.Loader | null {
   switch (mediaType) {
     case "JavaScript":
     case "Mjs":
@@ -54,7 +54,7 @@ export function mediaTypeToLoader(mediaType: MediaType): esbuild.Loader {
     case "Json":
       return "json";
     default:
-      throw new Error(`Unhandled media type ${mediaType}.`);
+      return null;
   }
 }
 
@@ -242,7 +242,7 @@ function mapJsLikeExtension(
   }
 }
 
-function mediaTypeFromSpecifier(specifier: URL): MediaType {
+export function mediaTypeFromSpecifier(specifier: URL): MediaType {
   const path = specifier.pathname;
   switch (extname(path)) {
     case "":

--- a/testdata/hello.txt
+++ b/testdata/hello.txt
@@ -1,0 +1,1 @@
+Hello World!

